ner_v2/detectors/numeral/number/number_detection.py

```diff
@@ -143,12 +160,28 @@ def detect_entity(self, text, **kwargs):
             number_unit = number_value_dict[NUMBER_DETECTION_RETURN_DICT_UNIT]
             if self.min_digit <= self._num_digits(number_value) <= self.max_digit:
                 if self.unit_type and (number_unit is None or
-                                       self.language_number_detector.units_map[number_unit].type != self.unit_type)\
+                                       self.language_number_detector.units_map[number_unit].type != self.unit_type) \
                         and not self.detect_without_unit:
                     continue
                 validated_number.append(number_value_dict)
                 validated_number_text.append(original_text)
 
+        if not validated_number:
```

Comment:
Hey Tanaya, why did we add this logic, and can you explain a little bit about why we want to run it only when all of the number replacement suggestions failed.

Reply:
Suppose we get a `text="123 456"`  so the numbers detected are 123 and 456 separately,

but if min_digit = 6,
Then `self.min_digit <= self._num_digits(number_value) <= self.max_digit = FALSE` and thus we do not find anything.

In the case that no number is found such that it doesn't satisfy these conditions, we have this new logic where we concatenate those two numbers together to provide a singular number.

We don't want this to get triggered if 3 digits is an allowed condition.